src/graph/graph.test.ts

```diff
@@ -0,0 +1,88 @@
+import { describe, expect, it } from 'vitest';
+import { NAME_ATTRIBUTE } from '../core/ids/system.js';
+import type { Op } from '../types.js';
+import { createEntity } from './create-entity.js';
+import { createRelationType } from './create-relation-type.js';
+import { createType } from './create-type.js';
+
+describe('Graph', () => {
+  it('creates multiple types, relation types, entities, relations', async () => {
+    const operations: Array<Op> = [];
+
+    // create person type
+    const {
+      id: personTypeId,
+      ops: createPersonTypeOps,
+      mapping: personMapping,
+    } = await createType({
+      attributes: {
+        name: {
+          id: NAME_ATTRIBUTE,
+          type: 'TEXT',
+          name: 'Name',
+        },
+        age: {
+          // ID is optional and will be generated automatically
+          type: 'NUMBER',
+          name: 'Age',
+        },
+      },
+    });
+    operations.push(...createPersonTypeOps);
+
+    // create restaurant type
+    const {
+      id: restaurantTypeId,
+      ops: createRestaurantTypeOps,
+      mapping: restaurantMapping,
+    } = await createType({
+      attributes: {
+        name: {
+          id: NAME_ATTRIBUTE,
+          type: 'TEXT',
+          name: 'Name',
+        },
+      },
+    });
+    operations.push(...createRestaurantTypeOps);
+
+    // create loves relation type
+    const { id: relationTypeId, ops: createRelationTypeOps } = await createRelationType({
+      from: personTypeId,
+      to: restaurantTypeId,
+      name: 'Loves',
+    });
+    operations.push(...createRelationTypeOps);
+
+    // create restaurant entity
+    const { id: restaurantId, ops: createRestaurantOps } = await createEntity({
+      types: [restaurantTypeId],
+      attributes: {
+        name: 'Yummy Dining',
+      },
+      mapping: restaurantMapping,
+    });
+    operations.push(...createRestaurantOps);
+
+    // create person entity
+    const { id: personId, ops: createPersonOps } = await createEntity({
+      types: [personTypeId],
+      attributes: {
+        name: 'John Doe',
+      },
+      mapping: personMapping,
+      relations: [{ relationTypeId, toId: restaurantId }],
```

Comment:
I'm not sure about having Id on the keys here. Would think that relations should each look like:
{
  relationType,
  to,
  attributes,
  relations,
}

May want to show what this looks like with more than one relation (should have separate key, vals). I assume that the from is implied here as being the entity we're creating. Think it's worth making sure we can support adding additional properties to the relations.